In [1]:
import pandas as pd    
import numpy as np   
import matplotlib.pyplot as plt  
import seaborn as sns 
        

In [2]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')
internet_use = pd.read_csv('../data/internet_use.csv', nrows=4497)

In [3]:
gdp_df.head()


Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN

In [4]:
gdp_df.tail()

Country or Area  Year        Value  Value Footnotes
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN

In [5]:
gdp_df.shape

(7176, 4)

In [6]:
print(gdp_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7176 entries, 0 to 7175
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7176 non-null   object 
 1   Year             7176 non-null   int64  
 2   Value            7176 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 224.4+ KB
None


QUESTION 5. How many rows and columns are in `gdp_df`? ----  gdp_df has 7176 rows, 4 columns.

What are the data types of each column? ---- Country or Area - object, 
Year - int64, Value - float64, Value Footnotes - float64


In [7]:
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2020  1970.560169              NaN
1        Afghanistan  2019  2065.036235              NaN
2        Afghanistan  2018  2033.804389              NaN
3        Afghanistan  2017  2058.400221              NaN
4        Afghanistan  2016  2057.067978              NaN
...              ...   ...          ...              ...
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN

[7176 rows x 4 columns]

In [8]:
gdp_df = gdp_df.drop('Value Footnotes', axis = 1)

In [9]:
gdp_df = gdp_df.rename(columns={'Country or Area': 'Country', 'Value' : 'GDP_Per_Capita'})

In [10]:
gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2020     1970.560169
1     Afghanistan  2019     2065.036235
2     Afghanistan  2018     2033.804389
3     Afghanistan  2017     2058.400221
4     Afghanistan  2016     2057.067978
...           ...   ...             ...
7171     Zimbabwe  1994     2603.927369
7172     Zimbabwe  1993     2420.092791
7173     Zimbabwe  1992     2437.152908
7174     Zimbabwe  1991     2733.696626
7175     Zimbabwe  1990     2652.129209

[7176 rows x 3 columns]

In [11]:
country_count = gdp_df.groupby('Country').count()
country_count

Year  GDP_Per_Capita
Country                                          
Afghanistan                    19              19
Africa Eastern and Southern    31              31
Africa Western and Central     31              31
Albania                        31              31
Algeria                        31              31
...                           ...             ...
Vietnam                        31              31
West Bank and Gaza             27              27
World                          31              31
Zambia                         31              31
Zimbabwe                       31              31

[242 rows x 2 columns]

In [19]:
country_count.max()


Year              31
GDP_Per_Capita    31
dtype: int64

In [29]:
country_count == country_count.max()

Year  GDP_Per_Capita
Country                                           
Afghanistan                  False           False
Africa Eastern and Southern   True            True
Africa Western and Central    True            True
Albania                       True            True
Algeria                       True            True
...                            ...             ...
Vietnam                       True            True
West Bank and Gaza           False           False
World                         True            True
Zambia                        True            True
Zimbabwe                      True            True

[242 rows x 2 columns]

In [37]:
countries_with_data = (country_count == country_count.max()).sum()
countries_with_data

Year              205
GDP_Per_Capita    205
dtype: int64

In [45]:
countries_missing_data = (country_count != country_count.max())
print(countries_missing_data)

                              Year  GDP_Per_Capita
Country                                           
Afghanistan                   True            True
Africa Eastern and Southern  False           False
Africa Western and Central   False           False
Albania                      False           False
Algeria                      False           False
...                            ...             ...
Vietnam                      False           False
West Bank and Gaza            True            True
World                        False           False
Zambia                       False           False
Zimbabwe                     False           False

[242 rows x 2 columns]


QUESTION 7. How many countries have data for all years? 205

Which countries are missing many years of data?

Look at the number of observations per year. What do you notice? 